Formula/y/yass-cli.rb

```diff
@@ -0,0 +1,70 @@
+class YassCli < Formula
+  desc "Lightweight and efficient, socks/http forward proxy"
+  homepage "https://letshack.info"
+  url "https://github.com/Chilledheart/yass/releases/download/1.13.1/yass-1.13.1.tar.bz2"
+  sha256 "26015d32a6bdc5bf2670b36c8f5fb68f4a324d577e027bc5eef18ae977e00c9f"
+  license "GPL-2.0-only"
+  head "https://github.com/Chilledheart/yass.git", branch: "develop"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "cmake" => :build
+  depends_on "go" => :build
+  depends_on "llvm" => :build if DevelopmentTools.clang_build_version < 1500
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+
+  depends_on "c-ares"
+  depends_on "libnghttp2"
+  depends_on "mbedtls"
+
+  uses_from_macos "zlib"
+
+  on_linux do
+    depends_on "zlib"
+  end
+
+  def install
+    ENV.llvm_clang if OS.mac? && (DevelopmentTools.clang_build_version < 1500)
+
+    args = %w[
+      -DBUILD_SHARED_LIBS=off
+      -DUSE_BUILTIN_CA_BUNDLE_CRT=off
+      -DCLI=on
+      -DSERVER=off
+      -DGUI=off
+      -DBUILD_TESTS=off
+      -DUSE_SYSTEM_MBEDTLS=on
+      -DUSE_ZLIB=on
+      -DUSE_SYSTEM_ZLIB=on
+      -DUSE_CARES=on
+      -DUSE_SYSTEM_CARES=on
+      -DUSE_SYSTEM_NGHTTP2=on
+    ]
+
+    args << "-DUSE_LIBCXX=on" if OS.mac?
+
+    system "cmake", "-S", ".", "-B", "build", "-G", "Ninja", *(std_cmake_args + args)
+    system "cmake", "--build", "build"
+
+    bin.install "build/yass_cli"
+  end
+
+  def caveats
+    on_macos do
+      <<~EOS
+        This formula only installs the command-line utilities by default.
+
+        Install yass.app with Homebrew Cask:
+          brew install --cask yass
+      EOS
+    end
+  end
+
+  test do
+    assert_match "yass_cli #{version}", shell_output("#{bin}/yass_cli --version")
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
We can build unittest binary but it is unlikely to be shipped inside a bottle.

Reply:
We don't need a unit test binary. From the linked documentation:

> Some advice for specific cases:
>
> - If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream’s documentation / source examples. A good example is [tinyxml2](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/t/tinyxml2.rb)’s test, which writes a small C++ source file into the test directory, compiles and links it against the tinyxml2 library and finally checks that the resulting program runs successfully.
> - If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
> - If the software cannot function without credentials or requires a virtual machine, docker instance, etc. to run, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected. This is preferred over mocking a dependency.
> - Homebrew comes with a number of [standard test fixtures](https://github.com/Homebrew/brew/tree/master/Library/Homebrew/test/support/fixtures), including numerous sample images, sounds, and documents in various formats. You can get the file path to a test fixture with e.g. test_fixtures("test.svg").

See the linked documentation for more details.

Reply:
We can provide some configure validation feature inside cli tool just like the gui one(see homebrew-cask) later. But currently, version check is what we have now.

Reply:
Tracked in https://github.com/Chilledheart/yass/issues/1085

Reply:
We don't need a validation feature. We should just run a command some user might run using `yas_cli`.

Reply:
yes, then we should be able to run unittests just like Gentoo-like/FreeBSD ports-like system. It is much safer and give us more guarentee.

Reply:
Gentoo-like: https://github.com/microcai/gentoo-zh/blob/master/net-proxy/yass/yass-1.13.1.ebuild
FreeBSD ports like: https://bugs.freebsd.org/bugzilla/attachment.cgi?id=251559&action=edit